### PR TITLE
Update outdated runners in the benchmarks folder

### PR DESCRIPTION
**Pull request by gspencergoog**
## Description

This PR updates the platform runners in the benchmarks subdir.

I've attempted to replace the entire group with new code as if the project were recreated with `flutter create` so that they don't become a Frankenstein of minor changes to make it work again, but just plain old generated code that can be replaced and updated in place.

## Tests
 - No tests because we're just updating boilerplate here.

### CODE REVIEW

File: dev/benchmarks/complex_layout/android/build.gradle [hunk 1]
```diff
@@ -2,25 +2,17 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
-// This file is auto generated.
-// To update all the build.gradle files in the Flutter repo,
-// See dev/tools/bin/generate_gradle_lockfiles.dart.
```
Comment (blasten) [inline dev/benchmarks/complex_layout/android/build.gradle:7]: nit: this file is auto-generated

File: dev/benchmarks/complex_layout/android/settings.gradle [hunk 2]
```diff
@@ -2,14 +2,8 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
-// This file is auto generated.
-// To update all the settings.gradle files in the Flutter repo,
-// See dev/tools/bin/generate_gradle_lockfiles.dart.
```
Comment (blasten) [inline dev/benchmarks/complex_layout/android/settings.gradle:7]: ditto

File: dev/benchmarks/complex_layout/android/build.gradle [hunk 3]
```diff
@@ -2,25 +2,17 @@
...
-    ext.kotlin_version = '1.4.32'
+    ext.kotlin_version = '1.3.50'
     repositories {
         google()
         mavenCentral()
     }
 
     dependencies {
-        classpath 'com.android.tools.build:gradle:7.0.0'
+        classpath 'com.android.tools.build:gradle:4.1.0'
```
[the remark this answers is not in the document]
Comment (gspencergoog) [reply dev/benchmarks/complex_layout/android/build.gradle:13]: Ahh, OK, I'll revert this. Is someone migrating the template too?

**Comment by gspencergoog** [comment]
> Thinking this through, I wonder if we should have a more general purpose script that keeps the boilerplate in sync.

I think that's an excellent idea.

**Comment by gspencergoog** [comment]
In doing this, I found there were a few directories that needed to be exempt (there's a test for the outdated embeddings, and there's an android test where it has a custom runner, for instance), but in general I think that should be possible.

My basic methodology was to remove all of the boilerplate directories entirely, and then regenerate using `flutter create .` with the appropriate platforms and arguments, and then `git add` the new boilerplate directories.

**Comment by blasten** [comment]
got it.  We don't have a well defined pattern for testing older project files, or even showing good error messages in the tool when the current project files are out-dated.

**Comment by gspencergoog** [comment]
Once we have a migration script, we could have a CI test that runs the script and sees if there is any difference, though. That would at least catch the runners that are out of date in the repo when we update the templates.

File: dev/benchmarks/complex_layout/android/build.gradle (hunk 3 above)
[the remark this answers is not in the document]
Comment (gspencergoog) [reply dev/benchmarks/complex_layout/android/build.gradle]: Actually, I think I'll wait for #90642 to land, and then update this and the other PR. It's likely to conflict in other ways too, and I don't want make Taha have to merge even more files.

**Review by blasten — APPROVED** [review]

**Pull request merged**